src/tree/azureResourceNodeContributor.ts

```diff
@@ -0,0 +1,97 @@
+import * as k8s from 'vscode-kubernetes-tools-api';
+import * as vscode from 'vscode';
+import { Errorable } from '../commands/utils/errorable';
+
+export class AzureResourceNodeContributor implements k8s.ClusterExplorerV1.NodeContributor {
+    constructor(
+        private readonly explorer: k8s.ClusterExplorerV1,
+        private readonly kubectl: k8s.KubectlV1
+    ) {}
+
+    contributesChildren(parent?: k8s.ClusterExplorerV1.ClusterExplorerNode | undefined): boolean {
+        return parent?.nodeType === 'context';
+    }
+
+    async getChildren(parent: k8s.ClusterExplorerV1.ClusterExplorerNode | undefined): Promise<k8s.ClusterExplorerV1.Node[]> {
+        if (parent === undefined) {
+            return [];
+        }
+
+        return [new AzureServicesFolderNode(this.explorer, this.kubectl)];
+    }
+}
+
+class AzureServicesFolderNode implements k8s.ClusterExplorerV1.Node {
+    constructor(
+        private readonly explorer: k8s.ClusterExplorerV1,
+        private readonly kubectl: k8s.KubectlV1
+    ) {}
+
+    async getChildren(): Promise<k8s.ClusterExplorerV1.Node[]> {
+        const crdTypes = await getAzureServiceResourceTypes(this.kubectl);
+        if (!crdTypes.succeeded) {
+            vscode.window.showErrorMessage(`Unable to retrieve Azure Services resources: ${crdTypes.error}.`);
+            return [];
+        }
+
+        // Map the custom resource definitions into NodeSources that the K8s extension knows how to turn into nodes.
+        const nodeSources = crdTypes.result.map(t => this.explorer.nodeSources.resourceFolder(t.displayName, t.pluralDisplayName, t.manifestKind, t.abbreviation));
+
+        // Turn the NodeSources into nodes.
+        const nodes = (await Promise.all(nodeSources.map(ns => ns.nodes()))).flat();
```

Comment:
I am not sure programmatically getting the nodes from a `NodeContributor` and using them elsewhere works - I recall having this same idea and one point and finding some odd behaviour - but I can't substantiate that.  I am probably hallucinating (or was doing something else, like a fully custom contributor, or it was an earlier version of the API where these worked a bit differently).  But I'd encourage particularly thorough testing just in case.

Reply:
Thanks @itowlson - @Tatsinnit and I have been testing and it appears to hold up.

One thing we did notice is that the last two options (watch and stop watching) are missing:

<img width="68" alt="image" src="https://user-images.githubusercontent.com/1817696/178167184-5c0e2940-be1a-4788-8f14-03aa7c75cb70.png">

I'm not sure if that's anything at all to be concerned about, but thought I'd call it out in case it's related to the odd behaviour you were mentioning. My uninformed guess is it relates to the API version of the cluster explorer (v1 vs. v1.1 or something?) that we're using.

As it seems to function okay, we'll merge this for now, but please shout if anything sounds any alarm bells and I'll dig into it some more.

Reply:
Watch-related commands require a specific context string on the tree item, which is added here:

https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/blob/92582a57ce722e166d25482b443c37a58d16305a/src/components/clusterexplorer/explorer.ts#L118

Not sure why this wouldn't appear on the 'duplicated' nodes if it appears on the 'ordinary' ones.  But that's the line where I'd start looking.

Reply:
Forgot to say, I'm delighted this is working for you and thanks for all the feedback!

Reply:
Ahh, I think that makes sense. It's missing the `apiName` property because the v1 `resourceFolder` function doesn't take that as the last argument: [v1](https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/blob/92582a57ce722e166d25482b443c37a58d16305a/src/api/contract/cluster-explorer/v1.ts#L104) vs. [v1_1](https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/blob/92582a57ce722e166d25482b443c37a58d16305a/src/api/contract/cluster-explorer/v1_1.ts#L106).

Thank you for the hint, @itowlson!

Any thoughts about whether we should be adopting the newer API, @Tatsinnit? I have no idea what's involved in doing that...

Reply:
The 1.1 API should be fully backward compatible except for using the `V1_1` namespace, i.e. you shouldn't need to change anything except for where you're passing that new optional argument.  They both back onto identical implementation code so I'd be confident there would be no migration issues.